src/Interpreters/DDLWorker.cpp

```diff
@@ -1204,17 +1214,22 @@ void DDLWorker::runMainThread()
             subsequent_errors_count = 0;
 
             LOG_DEBUG(log, "Waiting for queue updates");
+            auto zookeeper = getAndSetZooKeeper();
             queue_updated_event->wait();
+            if (zookeeper->expired())
```

Comment:
It shouldn't be expired right after calling `getAndSetZooKeeper` (it still can be, for example when the new session got expired immediately after creating it for some reason, but checking `zookeeper->expired()` right after `getAndSetZooKeeper` looks strange)